pilot/pkg/config/kube/ior/route.go

```diff
@@ -328,24 +317,16 @@ func (r *route) SetNamespaces(namespaces ...string) {
 
 	// In the first update we perform an initial sync
 	go func() {
-		r.initialSyncLock.Lock()
-		initialSyncRun := r.initialSyncRun
-		r.initialSyncLock.Unlock()
-
-		if !initialSyncRun {
-			// But only after gateway store cache is synced
-			IORLog.Debug("Waiting for the Gateway store cache to sync before performing our initial sync")
-			cache.WaitForNamedCacheSync("Gateways", r.stop, r.store.HasSynced)
-			IORLog.Debug("Gateway store cache synced. Performing our initial sync now")
-
-			if err := r.initialSync(); err != nil {
-				IORLog.Errora(err)
-			}
+		// But only after gateway store cache is synced
+		IORLog.Debug("Waiting for the Gateway store cache to sync before performing our initial sync")
+		cache.WaitForNamedCacheSync("Gateways", r.stop, r.store.HasSynced)
```

Comment:
Shouldn't we be checking the return value here? If it returns false, it means the stop channel was closed and we probably don't want to proceed with the sync.

Reply:
@bison Done.